File: tools/EventClients/examples/c#/XBMCDemoClient1.cs
using System;
using System.Collections.Generic;
using System.IO;
using System.Windows.Forms;
using XBMC;

namespace XBMCEventClientDemo
{
    static class Program
    {
        /// <summary>
        /// The main entry point for the application.
        /// </summary>
        [STAThread]
        static void Main()
        {
            EventClient eventClient = new EventClient();
            eventClient.Connect("127.0.0.1", 9777);

            string iconFile = @"../../icons/icon.png";
            IconType iconType = IconType.ICON_PNG;

            if !File.Exists(iconFile) {
                iconFile = @"/usr/share/xbmc/media/icon.png";
                if !File.Exists(iconFile) {
                    iconType = IconType.ICON_NONE;
                }
            }

            eventClient.SendHelo("XBMC Client Demo", iconType, iconFile);
            System.Threading.Thread.Sleep(1000);
            eventClient.SendNotification("XBMC Client Demo", "Notification Message", iconType, iconFile);
            System.Threading.Thread.Sleep(1000);
            eventClient.SendButton("dpadup", "XG", ButtonFlagsType.BTN_DOWN | ButtonFlagsType.BTN_NO_REPEAT, 0);
            System.Threading.Thread.Sleep(1000);
            eventClient.SendPing();
            System.Threading.Thread.Sleep(1000);
            eventClient.SendMouse(32768, 32768);
            System.Threading.Thread.Sleep(1000);
            eventClient.SendLog(LogTypeEnum.LOGERROR, "Example error log message from XBMC Client Demo");
            System.Threading.Thread.Sleep(1000);
            eventClient.SendAction("Mute");
            System.Threading.Thread.Sleep(1000);
            eventClient.SendBye();
        }
    }
}

File: tools/EventClients/examples/java/XBMCDemoClient1.java
package org.xbmc.eventclient.demo;

import java.io.File;
import java.io.IOException;
import java.net.Inet4Address;
import java.net.InetAddress;

import org.xbmc.eventclient.XBMCClient;

/**
 * Simple Demo EventClient
 * @author Stefan Agner
 *
 */
public class XBMCDemoClient1 {

	/**
	 * Simple Demo EventClient
	 * @param args
	 */
	public static void main(String[] args) throws IOException, InterruptedException {
		InetAddress host = Inet4Address.getByAddress(new byte[] { (byte)127, 0, 0, 1 } );

		Thread.sleep(20000);

		String iconFile = "/usr/share/xbmc/media/icon.png";

		if (! new File(iconFile).exists()) {
			iconFile = "../../icons/icon.png";

			if (! new File(iconFile).exists()) {
				iconFile = "";
			}
		}

		XBMCClient oXBMCClient = null;

		if (iconFile != "") {
			oXBMCClient = new XBMCClient(host, 9777, "My Client", iconFile);
		} else {
			oXBMCClient = new XBMCClient(host, 9777, "My Client");
		}

		Thread.sleep(7000);

		oXBMCClient.sendNotification("My Title", "My Message");


		Thread.sleep(7000);

		oXBMCClient.sendButton("KB", "escape", false, true, false, (short)0 , (byte)0);


		Thread.sleep(7000);
		oXBMCClient.sendButton("KB", "escape", true, true, false, (short)0 , (byte)0);
		oXBMCClient.sendNotification("My Title", "Escape sent");

		Thread.sleep(1000);

		oXBMCClient.sendButton("KB", "escape", true, false, false, (short)0 , (byte)0);
		oXBMCClient.sendNotification("My Title", "Escape released");

		Thread.sleep(7000);
		oXBMCClient.sendLog((byte)0, "My Client disconnects....");
		oXBMCClient.sendNotification("My Title", "Client will disconnect");
		oXBMCClient.stopClient();

	}

}
